todo_app/tests/test_app.py

```diff
@@ -0,0 +1,112 @@
+import os
+import pytest
+import requests
+from todo_app import app
+from dotenv import load_dotenv, find_dotenv
+
```

Comment:
In response to the question about checking the stub was called, we could write ourselves some helper functions like the below:
```python

stubs_called = None

def record(func):
    global stubs_called
    @functools.wraps(func)
    def wrapper(*args, **kwargs):
        stubs_called.append({
            'stub_name': func.__name__,
            'url': args[0],
            'kwargs': kwargs,
        })
        return func(*args, **kwargs)
    return wrapper

@pytest.fixture(autouse=True)
def reset_stubs_before_each_test():
    global stubs_called
    stubs_called = []
```

We can then set the stubs to record like so:
```python
@record
def stub(url, params={}):
```

And we can check the calls in our tests, something like:
```python
    assert len(stubs_called) == 1
    assert stubs_called[0]['url'] == "whatever"
    assert stubs_called[0]['kwargs']['data'] == "whatever"
    # highlight whatever you think is key
```